src/main/java/de/ait/secondlife/controllers/OfferController.java

```diff
@@ -0,0 +1,211 @@
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = OfferResponseWithPaginationDto.class))),
+    })
+    public ResponseEntity<OfferResponseWithPaginationDto> getAll(
+            @RequestParam(defaultValue = "0")
+            @Parameter(description = "Requested page number. ", example = "0")
+            int page,
+            @RequestParam(defaultValue = "10")
+            @Parameter(description = "Number of entities per page. ", example = "0")
+            int size,
+            @RequestParam(defaultValue = "createdAt")
+            @Parameter(description = "Scoring field. ", example = "createdAt")
+            String sortBy) {
+
+        Pageable pageable;
+        try {
+            pageable = PageRequest.of(page, size, Sort.by(sortBy).ascending());
+        } catch (IllegalArgumentException e) {
+            throw new PaginationParameterIsWrongException(page, size, sortBy);
+        }
+
+        return ResponseEntity.ok(service.findOffers(pageable));
+    }
+
+    @GetMapping("/{id}")
+    @Operation(
+            summary = "Get  offer by id",
+            description = "Receiving  offer by id available in the database "
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = OfferResponseDto.class))),
+    })
+    public ResponseEntity<OfferResponseDto> getById(
+            @PathVariable
+            @Parameter(description = "Offer id in UUID format.", example = "898449f7-e9d1-4d00-9fd6-cae203452f3a")
+            UUID id) {
+
+        return ResponseEntity.ok(service.findOfferById(id));
+    }
+
+
+    @GetMapping("/user/{id}")
+    @Operation(
+            summary = "Get all offer by user id",
+            description = "Receiving  all offer by user id available in the database with pagination"
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = OfferResponseWithPaginationDto.class))),
+    })
+    public ResponseEntity<OfferResponseWithPaginationDto> getByUserId(
+            @RequestParam(defaultValue = "0")
+            @Parameter(description = "Requested page number. ", example = "0")
+            int page,
+            @RequestParam(defaultValue = "10")
+            @Parameter(description = "Number of entities per page. ", example = "0")
+            int size,
+            @RequestParam(defaultValue = "createdAt")
+            @Parameter(description = "Scoring field. ", example = "createdAt")
```

Comment:
Is this "Sorting field" or something else?

Reply:
field by which sorting is performed